### ISSUE TITLE
'Save failed. Please try again.' when creating rooms on official iOS Hue app

### ISSUE THREAD

**Issue by caedn** [bwssytems/ha-bridge#256]
I'm running ha-bridge 3.5.1 with sudo on macOS Sierra 10.12.1. I've set it up as a proxy for my first gen Hue bridge and added 3 lights (passthru/hue) plus a virtual device (exec/Encapsulated).

Test ON/DIM/OFF in the web interface works great. As does controlling it via [hue-cli](https://github.com/bahamas10/hue-cli).

The problem occurs when I try to create a new room in the official Hue app (the new one with the colour in the logo). I can select the virtual bridge (manual ip) and get the list of my 4 devices but when I select one and hit save I get the rather basic error message `Save failed. Please try again.`. 

This shows up in the log (debug level) after hitting save:

    2016-11-20 21:05:15,289 [qtp719675491-16] DEBUG spark.webserver.MatcherFilter -httpMethod:post, uri: /api/851d5f1c8ca64d389ea833fb30d947de/groups
    2016-11-20 21:05:15,289 [qtp719675491-16] DEBUG spark.Request - get params
    2016-11-20 21:05:15,289 [qtp719675491-16] DEBUG spark.Request - get splat
    2016-11-20 21:05:15,290 [qtp719675491-16] DEBUG com.bwssystems.HABridge.hue.HueMulator - HH trace: hue api user create requested for device type: <not given> and username: 9359ed30583f447f854f2ecc142b6513

Creating rooms works fine when using the real Bridge.

[habridge.config](https://gist.github.com/anonymous/079bdceb53c5eb166f841744eea08622)
[device.db](https://gist.github.com/anonymous/a66429a7e7068137c72eba3114f6a59f)

`10.0.0.81` is my Hue Bridge, `.54` the Mac running ha-bridge.

**Comment by audiofreak9** [comment]
It's not a problem with your ha-bridge.  It discovers as a Hue Bridge but it is not coded to save rooms like a real Hue Bridge.  Future enhancement possibility...

**Comment by caedn** [comment]
I see. I assumed rooms/scenes were stored locally, thinking about it, saving it on the bridge does make a lot more sense. 

Thanks for the prompt reply.

**Comment by fauna5** [comment]
We'd need someone with a hue bridge to record the messages and then it
could be coded up.

Making it work with the hue app would be a cool test feature when setting
up the bridge.

**Comment by bwssytems** [comment]
As this bridge is to just emulate what the Echo can call, I was not worrying about the features of a real hue bridge. The intent of the bridge is to consolidate control to a single interface. The setup for each endpoint should be natively.

**Comment by caedn** [comment]
> The intent of the bridge is to consolidate control to a single interface. 
 
I was planning to add a few Pi based 433Mhz RF sockets and an Arduino based relay to habridge and control everything from the official Hue app to make use of iOS widgets. 

What is everyone else using to control the consolidated devices beside the Alexa products? Some third party app or the web interface? Or is my use case simply out of scope?

> The setup for each endpoint should be natively.

Sorry, I'm not sure I understand what you mean by that.

**Comment by bwssytems** [comment]
The original intent was to give the echo control over as many devices as possible with only the api calls that it needed (On, Off and Dim). The idea is you could front end this bridge with any home automation that supports a philips hue. I did not want to complicate the bridge to have full home automation features, such as room assignment and other items so as to keep it simple and talk to as many types of devices possible.

For now, your use case is out of scope.

**Comment by audiofreak9** [comment]
@caedn: I posted an [HA-Bridge Control](https://github.com/audiofreak9/HA-Bridge_Control) repository to github that I wrote to control my devices, it's PHP driven.  Follow the instructions to clone and install.  

It uses twitter bootstrap for mobile compatibility and fluidity so it can be used with a phone, tablet or desktop.  

I included a DIM slider, "Power" icon action buttons, and "Wrench" icon buttons to update ha-bridge device state only.
